dast/diff.py

```diff
@@ -0,0 +1,58 @@
+"""
+The main logic for the diff as it differs from DeepDiff
+"""
+from deepdiff import DeepDiff, DeepHash
+from deepdiff.diff import ListItemRemovedOrAdded, DiffLevel
+
+from dast.lcs import get_lcs
+
+class Diff(DeepDiff):
+    def _get_matching_pairs(self, level):
+        """
+        Given a level get matching pairs. This returns list of two tuples in the form:
+        [
+          (t1 index, t2 index), (t1 item, t2 item)
+        ]
+
+        Will compute the LCS between the two levels by using the DistanceMixin and a maximum cutoff threshold
+        """
+        t1_hashes = DeepHash(level.t1)
+        t1 = [t1_hashes[level.t1[i]] for i in range(len(level.t1))]
+        t2_hashes = DeepHash(level.t2)
+        t2 = [t2_hashes[level.t2[i]] for i in range(len(level.t2))]
+
+        lcs, t1_in_lcs, t2_in_lcs = get_lcs(t1, t2)
```

Comment:
I'm not actually using `t1_in_lcs` here and then just doing the logic underneath. I think we should maybe just remove it so `get_lcs` can remain nice and elegant (no extra bloat)

Reply:
Is it already nuke o'clock? :heart_eyes:

Reply:
Nuked